### PR TITLE
Open source canvas tests from flutter_web_ui

### PR DESCRIPTION
**Pull request by ditman**
Reimplement some painter tests as canvas screenshot tests in the engine.

Needs https://github.com/flutter/goldens/pull/53
Fixes https://github.com/flutter/flutter/issues/42027

(Set to WIP because I need to get the goldens pushed, then update the hash here)

### CODE REVIEW

**Comment by ditman** [comment]
`ERROR: Some files have trailing spaces. To fix, try something like `find . -name *.dart -exec sed -i -e 's/\s\+$//' {} \;`.` 🤦‍♂

File: lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (yjbanov) [inline lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart]: Let's not pub "scuba" in the names of new tests. Let's use the word "golden" instead. I'm also planning to rename the existing tests.

Comment (ditman) [reply lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart]: Fair, I'll rename them in the next rev (alongside adding the latest hash for goldens)

Comment (ditman) [reply lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart]: _(Renamed everything under engine, with git mv)_

**Comment by ditman** [comment]
Force-pushed after rebasing the branch

File: lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/test/golden_tests/engine/canvas_arc_scuba_test.dart]: :+1:

**Pull request merged**